include/zephyr/dt-bindings/clock/stm32f10x_clock.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (C) 2024, Joakim Andersson
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32F1_CLOCK_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32F1_CLOCK_H_
+
+#include "stm32_common_clocks.h"
+
+/** Domain clocks */
+
+/** Bus clocks */
+#define STM32_CLOCK_BUS_AHB1    0x014
+#define STM32_CLOCK_BUS_APB2    0x018
+#define STM32_CLOCK_BUS_APB1    0x01c
+
+#define STM32_PERIPH_BUS_MIN	STM32_CLOCK_BUS_AHB1
+#define STM32_PERIPH_BUS_MAX	STM32_CLOCK_BUS_APB1
+
+/** System clock */
+/* defined in stm32_common_clocks.h */
+
+/** Fixed clocks  */
+/* Low speed clocks defined in stm32_common_clocks.h */
+#define STM32_SRC_HSI               (STM32_SRC_LSI + 1)
+#define STM32_SRC_HSE               (STM32_SRC_HSI + 1)
+#define STM32_SRC_PLLCLK_DIV2       (STM32_SRC_HSE + 1)
+#define STM32_SRC_EXT_HSE           (STM32_SRC_PLLCLK_DIV2 + 1)
+#define STM32_SRC_PLL2CLK           (STM32_SRC_EXT_HSE + 1)
+#define STM32_SRC_PLLI2SCLK         (STM32_SRC_PLL2CLK + 1)
+#define STM32_SRC_PLLI2SCLK_DIV2    (STM32_SRC_PLLI2SCLK + 1)
```

Comment:
I added these based on Kconfig, but I cannot seem to find a reference manual that lists these for MCO select field.
Are these duplications of other definitions used?

These pairs look similar.

PLLI2SCLK
PLLI2S

PLL2PCLK
PLL2CLK

HSE
EXT_HSE

The _DIV2 I understand are special for the f1x / f10x lines as they don't have the PRESCALER
It may look like the MCO is always divided by two, and so maybe we don't need the _DIV2 to check that the clock source is enabled?

These definitions were added to make the macro expansion solution work properly, but we're not using that approach anymore.

Reply:
Going one by one:
* `PLLI2SCLK` is a misnomer for `PLL3` that (presumably) originated from STM32CubeF1.

https://github.com/zephyrproject-rtos/hal_stm32/blob/808602c497998f9e84f042ef90cf2010d69789dd/stm32cube/stm32f1xx/drivers/include/stm32f1xx_ll_rcc.h#L250-L272
```c
#define LL_RCC_MCO1SOURCE_PLLI2SCLK_DIV2   RCC_CFGR_MCO_PLL3CLK_DIV2 /*!< PLLI2S clock divided by 2 selected as MCO source*/
#define LL_RCC_MCO1SOURCE_PLLI2SCLK        RCC_CFGR_MCO_PLL3CLK      /*!< PLLI2S clock selected as MCO source */
```

Since we're dropping LL macros, I think it would be wiser to match RM and use `STM32_SRC_PLL3` - as it doesn't exist yet, support for it must be added.

* `PLLI2S` is a real PLL block on STM32F4/STM32F7

RM0368 makes it clear that MCO corresponds to PLLI2S R output, so the existing `STM32_SRC_PLLI2S_R` can be used for F4.

An identically named  macro should be introduced for F7 so it can be used there as well (RM0431 is *less explicit* about it, but `PLLI2S` MCO source should be `R` output on F7 too). 

* `PLL2PCLK` is `PLL2` `P` output on STM32H5/H7, so it can be mapped to existing `STM32_SRC_PLL2_P`

* `PLL2CLK` is the (single) `PLL2` output on STM32F1, so it should be mapped to a separate `STM32_SRC_PLL2` macro, for which support would need to be added.

* For the `_DIV2` sources, since we are not allowing `clock_control_get_subsys_rate` queries on the MCO sources for now, the regular and `_DIV2` sources can use the same `STM32_SRC_xxx` macro.
  * For `STM32_SRC_PLLCLK_DIV2` specifically, since no non-`_DIV2` equivalent exists, I think the macro can remain as `STM32_SRC_PLLCLK_DIV2` though.

Reply:
If I build for one of the connectivity lines i get the following options for MCO1 output:
`west build -b stm3210c_eval samples/boards/stm32/mco -t menuconfig`
```
(X) NOCLOCK
( ) EXT_HSE
( ) HSE
( ) HSI
( ) PLLCLK_DIV2
( ) PLL2CLK
( ) PLLI2SCLK
( ) PLLI2SCLK_DIV2
( ) SYSCLK
```

However looking up STM32F107VC SoC gives RM008 as the reference manual, which only has these options as MCOSEL:
```
SYSCLK
HSI
HSE
PLL clock divided by 2
```

So either I don't have the right reference manual, or the Kconfig selection for this board is wrong.

Reply:
The STM32F107VC is a device of the Connectivity Line (`Connectivity line devices are STM32F105xx and STM32F107xx microcontrollers.`). Are you sure that you looked at Chapter 8 "`Connectivity line devices: RCC`" of RM0008, and not Chapter 7 "`Low-, medium-, high- and XL-density RCC`"?

Quoting page 135 of RM0008 Rev 21 (corresponding Kconfig in `[]`):
```
Bits 26:24 MCO[3:0]: Microcontroller clock output
Set and cleared by software.
00xx: No clock                                                        [NOCLOCK]
0100: System clock (SYSCLK) selected                                  [SYSCLK]
0101: HSI clock selected                                              [HSI]
0110: HSE clock selected                                              [HSE]
0111: PLL clock divided by 2 selected                                 [PLLCLK_DIV2]
1000: PLL2 clock selected                                             [PLL2CLK]
1001: PLL3 clock divided by 2 selected                                [PLLI2SCLK_DIV2]
1010: XT1 external 3-25 MHz oscillator clock selected (for Ethernet)  [EXT_HSE]
1011: PLL3 clock selected (for Ethernet)                              [PLLI2SCLK]
```

Reply:
Thanks. I didn't know it was two different chapters for the RCC configuration. Then it makes sense.

Reply:
So, from what I can tell, what is missing is:
F10x (connectivity line) support for PLL3 clock source.
F7 support for PLLI2S_R clock source.

I will add:
PLL2 and PLL3 support to `clock_stm32_ll_common.c:enabled_clock()` function, and PLL2/3 clock source definition to f10x, and PLLI2S_R to f7.

At that point the DTS approach will be feature-complete compared to what the Kconfig approach had, so deprecation can happen, and the missing support can be added as a follow-up to this PR.

Reply:
> At that point the DTS approach will be feature-complete compared to what the Kconfig approach had, so deprecation can happen, and the missing support can be added as a follow-up to this PR (by someone else)

Excellent. Thanks again @joerchan

Reply:
Adding STM32_SRC_PLLCLK to stm32f1 devices cannot be done without also implementing `get_pllout_frequency` in `clock_stm32f1.c`.
However this seems to be the right way to refer to this node.

Is there an easy way to tell the frequency for this PLL? (looking at clock_f0_f3.c it looks a bit complicated, and like it needs to be a variant of that function).